### PR TITLE
Kotlinize the playground demo app

### PR DESCRIPTION
**Pull request by thagikura**

### CODE REVIEW

File: demo-playground/src/main/java/com/google/android/flexbox/FragmentHelper.kt [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+        })
+    }
+
+    private fun initializeJustifyContentSpinner(navigationMenu: Menu) {
+        initializeSpinner(flexContainer.justifyContent, R.id.menu_item_justify_content,
+                navigationMenu, R.array.array_justify_content,
+                object : AdapterView.OnItemSelectedListener {
+                    override fun onItemSelected(parent: AdapterView<*>, ignored: View?, position: Int,
+                                                id: Long) {
+                        val selected = parent.getItemAtPosition(position).toString()
```
Comment (MGaetan89) [inline demo-playground/src/main/java/com/google/android/flexbox/FragmentHelper.kt]:
```kotlin
flexContainer.justifyContent = when (parent.getItemAtPosition(position).toString()) {
    FLEX_START -> JustifyContent.FLEX_START
    FLEX_END -> JustifyContent.FLEX_END
    CENTER -> JustifyContent.CENTER
    SPACE_BETWEEN -> JustifyContent.SPACE_BETWEEN
    SPACE_AROUND -> JustifyContent.SPACE_AROUND
    else -> return
}
```

File: demo-playground/src/main/java/com/google/android/flexbox/FragmentHelper.kt [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+        })
+    }
+
+    private fun initializeAlignItemsSpinner(navigationMenu: Menu) {
+        initializeSpinner(flexContainer.alignItems, R.id.menu_item_align_items,
+                navigationMenu, R.array.array_align_items,
+                object : AdapterView.OnItemSelectedListener {
+                    override fun onItemSelected(parent: AdapterView<*>, ignored: View?, position: Int,
+                                                id: Long) {
+                        val selected = parent.getItemAtPosition(position).toString()
```
Comment (MGaetan89) [inline demo-playground/src/main/java/com/google/android/flexbox/FragmentHelper.kt]:
```kotlin
flexContainer.alignItems = when (parent.getItemAtPosition(position).toString()) {
    FLEX_START -> AlignItems.FLEX_START
    FLEX_END -> AlignItems.FLEX_END
    CENTER -> AlignItems.CENTER
    BASELINE -> AlignItems.BASELINE
    STRETCH -> AlignItems.STRETCH
    else -> return
}
```

File: demo-playground/src/main/java/com/google/android/flexbox/FragmentHelper.kt [hunk 3]
```diff
@@ -0,0 +1,341 @@
...
+                navigationMenu, R.array.array_align_content,
+                object : AdapterView.OnItemSelectedListener {
+                    override fun onItemSelected(parent: AdapterView<*>, ignored: View?, position: Int,
+                                                id: Long) {
+                        if (flexContainer is FlexboxLayoutManager) {
+                            Toast.makeText(activity, R.string.align_content_not_supported,
+                                    Toast.LENGTH_SHORT).show()
+                            return
+                        }
+                        val selected = parent.getItemAtPosition(position).toString()
```
Comment (MGaetan89) [inline demo-playground/src/main/java/com/google/android/flexbox/FragmentHelper.kt]:
```kotlin
flexContainer.alignContent = when (parent.getItemAtPosition(position).toString()) {
    FLEX_START -> AlignContent.FLEX_START
    FLEX_END -> AlignContent.FLEX_END
    CENTER -> AlignContent.CENTER
    SPACE_BETWEEN -> AlignContent.SPACE_BETWEEN
    SPACE_AROUND -> AlignContent.SPACE_AROUND
    STRETCH -> AlignContent.STRETCH
    else -> return
}
```

File: demo-playground/src/main/java/com/google/android/flexbox/FragmentHelper.kt [hunk 4]
```diff
@@ -0,0 +1,341 @@
...
+     * Converter for converting an int value for Flexbox properties to a String.
+     */
+    private interface ValueToStringConverter {
+
+        fun asString(value: Int): String
+    }
+
+    companion object {
+
+        private val DEFAULT_WIDTH = "120"
```
Comment (MGaetan89) [inline demo-playground/src/main/java/com/google/android/flexbox/FragmentHelper.kt]: `private const val`

File: demo-playground/src/main/java/com/google/android/flexbox/FragmentHelper.kt [hunk 5]
```diff
@@ -0,0 +1,341 @@
...
+
+        private val DEFAULT_HEIGHT = "80"
```
Comment (MGaetan89) [inline demo-playground/src/main/java/com/google/android/flexbox/FragmentHelper.kt]: `private const val`

File: demo-playground/src/main/java/com/google/android/flexbox/MainActivity.kt [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+            val intent = Intent(this, SettingsActivity::class.java)
+            startActivity(intent)
+            return true
+        }
+        return super.onOptionsItemSelected(item)
+    }
+
+    companion object {
+
+        private val FLEXBOXLAYOUT_FRAGMENT = "flexboxlayout_fragment"
```
Comment (MGaetan89) [inline demo-playground/src/main/java/com/google/android/flexbox/MainActivity.kt]: `private const val`

File: demo-playground/src/main/java/com/google/android/flexbox/MainActivity.kt [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+
+        private val RECYCLERVIEW_FRAGMENT = "recyclerview_fragment"
```
Comment (MGaetan89) [inline demo-playground/src/main/java/com/google/android/flexbox/MainActivity.kt]: `private const val`

File: demo-playground/src/main/java/com/google/android/flexbox/RecyclerViewFragment.kt [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+    }
+
+    override fun onSaveInstanceState(outState: Bundle) {
+        super.onSaveInstanceState(outState)
+        outState.putParcelableArrayList(FLEX_ITEMS_KEY, ArrayList(adapter.items))
+    }
+
+    companion object {
+
+        private val FLEX_ITEMS_KEY = "flex_items"
```
Comment (MGaetan89) [inline demo-playground/src/main/java/com/google/android/flexbox/RecyclerViewFragment.kt]: `private const val`

File: demo-playground/src/main/java/com/google/android/flexbox/Util.kt [hunk 9]
```diff
@@ -14,40 +14,43 @@
...
 
-import android.content.Context;
-import android.util.DisplayMetrics;
+import android.content.Context
 
 /**
  * Utilities class.
  */
-public class Util {
+internal object Util {
```
Comment (MGaetan89) [inline demo-playground/src/main/java/com/google/android/flexbox/Util.kt]: Those two methods could be extensions functions on `Context`.

File: demo-playground/src/main/java/com/google/android/flexbox/validators/DimensionInputValidator.kt [hunk 10]
```diff
@@ -14,21 +14,19 @@
...
 
-    @Override
-    public boolean isValidInput(CharSequence charSequence) {
+    override fun isValidInput(charSequence: CharSequence): Boolean {
         // -1 represents match_parent, -2 represents wrap_content
-        return !TextUtils.isEmpty(charSequence) &&
-                (TextUtils.isDigitsOnly(charSequence) ||
-                        charSequence.toString().equals("-1") ||
-                        charSequence.toString().equals("-2"));
+        return !TextUtils.isEmpty(charSequence) && (TextUtils.isDigitsOnly(charSequence) ||
```
Comment (MGaetan89) [inline demo-playground/src/main/java/com/google/android/flexbox/validators/DimensionInputValidator.kt]: `!TextUtils.isEmpty(charSequence)` => `!charSequence.isNullOrEmpty()`

File: demo-playground/src/main/java/com/google/android/flexbox/validators/FixedDimensionInputValidator.kt [hunk 11]
```diff
@@ -14,17 +14,16 @@
...
  * Validator for dimension values.
  */
-public class FixedDimensionInputValidator implements InputValidator {
+class FixedDimensionInputValidator : InputValidator {
 
-    @Override
-    public boolean isValidInput(CharSequence charSequence) {
-        return !TextUtils.isEmpty(charSequence) && TextUtils.isDigitsOnly(charSequence);
+    override fun isValidInput(charSequence: CharSequence): Boolean {
+        return !TextUtils.isEmpty(charSequence) && TextUtils.isDigitsOnly(charSequence)
```
Comment (MGaetan89) [inline demo-playground/src/main/java/com/google/android/flexbox/validators/FixedDimensionInputValidator.kt]: `!TextUtils.isEmpty(charSequence)` => `!charSequence.isNullOrEmpty()`

File: demo-playground/src/main/java/com/google/android/flexbox/validators/FlexBasisPercentInputValidator.kt [hunk 12]
```diff
@@ -14,20 +14,17 @@
...
+class FlexBasisPercentInputValidator : InputValidator {
 
-    @Override
-    public boolean isValidInput(CharSequence charSequence) {
+    override fun isValidInput(charSequence: CharSequence): Boolean {
         // -1 represents not set
-        return !TextUtils.isEmpty(charSequence) &&
-                (TextUtils.isDigitsOnly(charSequence) ||
-                        charSequence.toString().equals("-1"));
+        return !TextUtils.isEmpty(charSequence) && (TextUtils.isDigitsOnly(charSequence) || charSequence.toString() == "-1")
```
Comment (MGaetan89) [inline demo-playground/src/main/java/com/google/android/flexbox/validators/FlexBasisPercentInputValidator.kt]: `!TextUtils.isEmpty(charSequence)` => `!charSequence.isNullOrEmpty()`

File: demo-playground/src/main/java/com/google/android/flexbox/validators/IntegerInputValidator.kt [hunk 13]
```diff
@@ -14,25 +14,27 @@
...
 
 /**
  * Validator for the integers.
  */
-public class IntegerInputValidator implements InputValidator {
+class IntegerInputValidator : InputValidator {
 
-    @Override
-    public boolean isValidInput(CharSequence charSequence) {
+    override fun isValidInput(charSequence: CharSequence): Boolean {
```
Comment (MGaetan89) [inline demo-playground/src/main/java/com/google/android/flexbox/validators/IntegerInputValidator.kt]: Replace the whole body with `return charSequence.toString().toIntOrNull() != null`

File: demo-playground/src/main/java/com/google/android/flexbox/validators/NonNegativeDecimalInputValidator.kt [hunk 14]
```diff
@@ -14,21 +14,23 @@
...
 
 /**
  * Validator for non negative integers.
  */
-public class NonNegativeDecimalInputValidator implements InputValidator {
+class NonNegativeDecimalInputValidator : InputValidator {
 
-    @Override
-    public boolean isValidInput(CharSequence charSequence) {
+    override fun isValidInput(charSequence: CharSequence): Boolean {
```
Comment (MGaetan89) [inline demo-playground/src/main/java/com/google/android/flexbox/validators/NonNegativeDecimalInputValidator.kt]: Replace the whole body with `return charSequence.toString().toFloatOrNull() ?: -1f >= 0`

File: demo-playground/build.gradle [hunk 15]
```diff
@@ -45,6 +46,7 @@ dependencies {
     compile "com.android.support:preference-v7:${rootProject.ext.supportLibVersion}"
     compile "com.android.support:preference-v14:${rootProject.ext.supportLibVersion}"
     compile "com.android.support:design:${rootProject.ext.supportLibVersion}"
+    compile "org.jetbrains.kotlin:kotlin-stdlib-jre7:$kotlinVersion"
```
Comment (thagikura) [inline demo-playground/build.gradle]: Done

File: demo-playground/src/main/java/com/google/android/flexbox/FlexItemAdapter.kt [hunk 16]
```diff
@@ -0,0 +1,68 @@
...
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import com.google.android.apps.flexbox.R
+
+/**
+ * [RecyclerView.Adapter] implementation for [FlexItemViewHolder].
+ */
+internal class FlexItemAdapter(private val activity: AppCompatActivity, private val flexContainer: FlexContainer) : RecyclerView.Adapter<FlexItemViewHolder>() {
+
+    private val layoutParams: MutableList<FlexboxLayoutManager.LayoutParams> = ArrayList()
```
Comment (thagikura) [inline demo-playground/src/main/java/com/google/android/flexbox/FlexItemAdapter.kt]: Done

File: demo-playground/src/main/java/com/google/android/flexbox/FlexItemAdapter.kt [hunk 17]
```diff
@@ -0,0 +1,68 @@
...
+    fun removeItem(position: Int) {
+        if (position < 0 || position >= layoutParams.size) {
+            return
+        }
+        layoutParams.removeAt(position)
+        notifyItemRemoved(layoutParams.size)
+        notifyItemRangeChanged(position, layoutParams.size)
+    }
+
+    val items: MutableList<FlexboxLayoutManager.LayoutParams>
```
Comment (thagikura) [inline demo-playground/src/main/java/com/google/android/flexbox/FlexItemAdapter.kt]: Done

File: demo-playground/src/main/java/com/google/android/flexbox/FlexItemAdapter.kt [hunk 18]
```diff
@@ -0,0 +1,68 @@
...
+        get() = layoutParams
+
+    override fun getItemCount(): Int {
```
Comment (thagikura) [inline demo-playground/src/main/java/com/google/android/flexbox/FlexItemAdapter.kt]: Done

File: demo-playground/src/main/java/com/google/android/flexbox/FlexItemEditFragment.kt [hunk 19]
```diff
@@ -0,0 +1,417 @@
...
+     */
+    private lateinit var flexItemInEdit: FlexItem
+
+    private var flexItemChangedListener: FlexItemChangedListener? = null
+
+    override fun onAttach(context: Context) {
+        super.onAttach(context)
+    }
+
+    override fun onDetach() {
```
Comment (thagikura) [inline demo-playground/src/main/java/com/google/android/flexbox/FlexItemEditFragment.kt]: Done

File: demo-playground/src/main/java/com/google/android/flexbox/FlexItemEditFragment.kt [hunk 20]
```diff
@@ -0,0 +1,417 @@
...
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+            setStyle(DialogFragment.STYLE_NORMAL, android.R.style.Theme_Material_Light_Dialog)
+        } else if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.HONEYCOMB) {
+            setStyle(DialogFragment.STYLE_NORMAL, android.R.style.Theme_Holo_Light_Dialog)
+        } else {
+            setStyle(DialogFragment.STYLE_NORMAL, android.R.style.Theme_Dialog)
+        }
+        val args = arguments
```
Comment (thagikura) [inline demo-playground/src/main/java/com/google/android/flexbox/FlexItemEditFragment.kt]: Done

File: demo-playground/src/main/java/com/google/android/flexbox/FlexItemEditFragment.kt [hunk 21]
```diff
@@ -0,0 +1,417 @@
...
+                widthEdit, heightEdit, minWidthEdit, minHeightEdit, maxWidthEdit, maxHeightEdit)
+
+        val alignSelfSpinner = view.findViewById(
+                R.id.spinner_align_self) as Spinner
+        val arrayAdapter = ArrayAdapter.createFromResource(activity,
+                R.array.array_align_self, R.layout.spinner_item)
+        alignSelfSpinner.adapter = arrayAdapter
+        alignSelfSpinner.onItemSelectedListener = object : AdapterView.OnItemSelectedListener {
+            override fun onItemSelected(parent: AdapterView<*>, ignored: View, position: Int, id: Long) {
+                val selected = parent.getItemAtPosition(position).toString()
```
Comment (thagikura) [inline demo-playground/src/main/java/com/google/android/flexbox/FlexItemEditFragment.kt]: Done

File: demo-playground/src/main/java/com/google/android/flexbox/FlexItemEditFragment.kt [hunk 22]
```diff
@@ -0,0 +1,417 @@
...
+                textInputLayout.error = activity.resources.getString(errorMessageId)
+            }
+        }
+
+        override fun afterTextChanged(editable: Editable) {
+            if (textInputLayout.isErrorEnabled || TextUtils.isEmpty(editable) ||
+                    !inputValidator.isValidInput(editable.toString())) {
+                return
+            }
+            val value: Number
```
Comment (thagikura) [inline demo-playground/src/main/java/com/google/android/flexbox/FlexItemEditFragment.kt]: Couldn't find a good way of mixing Int and Float in a single variable.
Instead parsed as Float first, then cast to Int later if needed. Please let me know if there is better way

File: demo-playground/src/main/java/com/google/android/flexbox/FlexItemEditFragment.kt [hunk 23]
```diff
@@ -0,0 +1,417 @@
...
+
+            when (textInputLayout.id) {
+                R.id.input_layout_order -> if (flexItemInEdit !is FlexboxLayoutManager.LayoutParams) {
+                    flexItemInEdit.order = value.toInt()
+                }
+                R.id.input_layout_flex_grow -> flexItemInEdit.flexGrow = value.toFloat()
+                R.id.input_layout_flex_shrink -> flexItemInEdit.flexShrink = value.toFloat()
+                R.id.input_layout_width -> flexItemInEdit.width = Util.dpToPixel(context, value.toInt())
+                R.id.input_layout_height -> flexItemInEdit.height = Util.dpToPixel(context, value.toInt())
+                R.id.input_layout_flex_basis_percent -> if (value.toInt().toFloat() != FlexboxLayout.LayoutParams.FLEX_BASIS_PERCENT_DEFAULT) {
```
Comment (thagikura) [inline demo-playground/src/main/java/com/google/android/flexbox/FlexItemEditFragment.kt]: Done

File: demo-playground/src/main/java/com/google/android/flexbox/FlexItemEditFragment.kt [hunk 24]
```diff
@@ -0,0 +1,417 @@
...
+                    flexItemInEdit.flexBasisPercent = (value.toInt() / 100.0).toFloat()
```
Comment (thagikura) [inline demo-playground/src/main/java/com/google/android/flexbox/FlexItemEditFragment.kt]: Done

File: demo-playground/src/main/java/com/google/android/flexbox/FlexItemEditFragment.kt [hunk 25]
```diff
@@ -0,0 +1,417 @@
...
+        to.minHeight = from.minHeight
+        to.maxWidth = from.maxWidth
+        to.minWidth = from.minWidth
+        to.alignSelf = from.alignSelf
+        to.isWrapBefore = from.isWrapBefore
+    }
+
+    companion object {
+
+        private val FLEX_ITEM_KEY = "flex_item"
```
Comment (thagikura) [inline demo-playground/src/main/java/com/google/android/flexbox/FlexItemEditFragment.kt]: Done

File: demo-playground/src/main/java/com/google/android/flexbox/FlexItemEditFragment.kt [hunk 26]
```diff
@@ -0,0 +1,417 @@
...
+
+        private val VIEW_INDEX_KEY = "view_index"
```
Comment (thagikura) [inline demo-playground/src/main/java/com/google/android/flexbox/FlexItemEditFragment.kt]: Done

File: demo-playground/src/main/java/com/google/android/flexbox/FlexboxLayoutFragment.kt [hunk 27]
```diff
@@ -0,0 +1,116 @@
...
+        return TextView(context).apply {
+            setBackgroundResource(R.drawable.flex_item_background)
+            text = (index + 1).toString()
+            gravity = Gravity.CENTER
+        }
+    }
+
+    companion object {
+
+        private val FLEX_ITEMS_KEY = "flex_items_key"
```
Comment (thagikura) [inline demo-playground/src/main/java/com/google/android/flexbox/FlexboxLayoutFragment.kt]: Done

File: demo-playground/src/main/java/com/google/android/flexbox/FragmentHelper.kt [hunk 28]
```diff
@@ -0,0 +1,341 @@
...
+        flexItem.flexGrow = readPreferenceAsFloat(activity.getString(R.string.new_flex_grow_key), "0.0")
+        flexItem.flexShrink = readPreferenceAsFloat(activity.getString(R.string.new_flex_shrink_key), "1.0")
+        val flexBasisPercent = readPreferenceAsInteger(
+                activity.getString(R.string.new_flex_basis_percent_key), "-1")
+        flexItem.flexBasisPercent = if (flexBasisPercent == -1) -1f else (flexBasisPercent / 100.0).toFloat()
+        return flexItem
+    }
+
+    private fun readPreferenceAsInteger(key: String, defValue: String): Int {
+        if (sharedPreferences.contains(key)) {
```
Comment (thagikura) [inline demo-playground/src/main/java/com/google/android/flexbox/FragmentHelper.kt]: Done

File: demo-playground/src/main/java/com/google/android/flexbox/FragmentHelper.kt [hunk 29]
```diff
@@ -0,0 +1,341 @@
...
+            return Integer.valueOf(sharedPreferences.getString(key, defValue))!!
+        } else {
+            return Integer.valueOf(defValue)!!
+        }
+    }
+
+    private fun readPreferenceAsFloat(key: String, defValue: String): Float {
+        if (sharedPreferences.contains(key)) {
```
Comment (thagikura) [inline demo-playground/src/main/java/com/google/android/flexbox/FragmentHelper.kt]: Done

File: demo-playground/src/main/java/com/google/android/flexbox/FragmentHelper.kt [hunk 30]
```diff
@@ -0,0 +1,341 @@
...
+        spinner.setSelection(position)
+    }
+
+    private fun initializeFlexDirectionSpinner(navigationMenu: Menu) {
+        initializeSpinner(flexContainer.flexDirection, R.id.menu_item_flex_direction,
+                navigationMenu, R.array.array_flex_direction,
+                object : AdapterView.OnItemSelectedListener {
+                    override fun onItemSelected(parent: AdapterView<*>, ignored: View?, position: Int,
+                                                id: Long) {
+                        val selected = parent.getItemAtPosition(position).toString()
```
Comment (thagikura) [inline demo-playground/src/main/java/com/google/android/flexbox/FragmentHelper.kt]: Done

File: demo-playground/src/main/java/com/google/android/flexbox/FragmentHelper.kt [hunk 31]
```diff
@@ -0,0 +1,341 @@
...
+        })
+    }
+
+    private fun initializeFlexWrapSpinner(navigationMenu: Menu) {
+        initializeSpinner(flexContainer.flexWrap, R.id.menu_item_flex_wrap,
+                navigationMenu, R.array.array_flex_wrap,
+                object : AdapterView.OnItemSelectedListener {
+                    override fun onItemSelected(parent: AdapterView<*>, ignored: View?, position: Int,
+                                                id: Long) {
+                        val selected = parent.getItemAtPosition(position).toString()
```
Comment (thagikura) [inline demo-playground/src/main/java/com/google/android/flexbox/FragmentHelper.kt]: Done

File: demo-playground/src/main/java/com/google/android/flexbox/FragmentHelper.kt (hunk 1 above)
Comment (thagikura) [inline demo-playground/src/main/java/com/google/android/flexbox/FragmentHelper.kt]: Done

File: demo-playground/src/main/java/com/google/android/flexbox/FragmentHelper.kt (hunk 2 above)
Comment (thagikura) [inline demo-playground/src/main/java/com/google/android/flexbox/FragmentHelper.kt]: Done

File: demo-playground/src/main/java/com/google/android/flexbox/FragmentHelper.kt (hunk 3 above)
Comment (thagikura) [inline demo-playground/src/main/java/com/google/android/flexbox/FragmentHelper.kt]: Done

File: demo-playground/src/main/java/com/google/android/flexbox/FragmentHelper.kt (hunk 4 above)
Comment (thagikura) [inline demo-playground/src/main/java/com/google/android/flexbox/FragmentHelper.kt]: Done

File: demo-playground/src/main/java/com/google/android/flexbox/FragmentHelper.kt (hunk 5 above)
Comment (thagikura) [inline demo-playground/src/main/java/com/google/android/flexbox/FragmentHelper.kt]: Done

File: demo-playground/src/main/java/com/google/android/flexbox/MainActivity.kt (hunk 6 above)
Comment (thagikura) [inline demo-playground/src/main/java/com/google/android/flexbox/MainActivity.kt]: Done

File: demo-playground/src/main/java/com/google/android/flexbox/MainActivity.kt (hunk 7 above)
Comment (thagikura) [inline demo-playground/src/main/java/com/google/android/flexbox/MainActivity.kt]: Done

File: demo-playground/src/main/java/com/google/android/flexbox/RecyclerViewFragment.kt (hunk 8 above)
Comment (thagikura) [inline demo-playground/src/main/java/com/google/android/flexbox/RecyclerViewFragment.kt]: Done

File: demo-playground/src/main/java/com/google/android/flexbox/Util.kt (hunk 9 above)
Comment (thagikura) [inline demo-playground/src/main/java/com/google/android/flexbox/Util.kt]: Done

File: demo-playground/src/main/java/com/google/android/flexbox/validators/DimensionInputValidator.kt (hunk 10 above)
Comment (thagikura) [inline demo-playground/src/main/java/com/google/android/flexbox/validators/DimensionInputValidator.kt]: Done

File: demo-playground/src/main/java/com/google/android/flexbox/validators/FixedDimensionInputValidator.kt (hunk 11 above)
Comment (thagikura) [inline demo-playground/src/main/java/com/google/android/flexbox/validators/FixedDimensionInputValidator.kt]: Done

File: demo-playground/src/main/java/com/google/android/flexbox/validators/FlexBasisPercentInputValidator.kt (hunk 12 above)
Comment (thagikura) [inline demo-playground/src/main/java/com/google/android/flexbox/validators/FlexBasisPercentInputValidator.kt]: Done

File: demo-playground/src/main/java/com/google/android/flexbox/validators/IntegerInputValidator.kt (hunk 13 above)
Comment (thagikura) [inline demo-playground/src/main/java/com/google/android/flexbox/validators/IntegerInputValidator.kt]: Done

File: demo-playground/src/main/java/com/google/android/flexbox/validators/NonNegativeDecimalInputValidator.kt (hunk 14 above)
Comment (thagikura) [inline demo-playground/src/main/java/com/google/android/flexbox/validators/NonNegativeDecimalInputValidator.kt]: Done

**Comment by thagikura** [comment]
Basically addressed all except for the String to Int and Float conversion in the FlexItemEditFragment.

Thanks!

File: demo-playground/src/main/java/com/google/android/flexbox/FlexItemClickListener.kt [hunk 32]
```diff
@@ -0,0 +1,40 @@
...
+ * the Flexbox Playground demo app.
+ */
+internal class FlexItemClickListener(private val activity: AppCompatActivity, private val flexItemChangedListener: FlexItemChangedListener,
+                                     private val viewIndex: Int) : View.OnClickListener {
+
+    override fun onClick(v: View) {
+        val fragment = FlexItemEditFragment
+                .newInstance(v.layoutParams as FlexItem, viewIndex)
+        fragment.setFlexItemChangedListener(flexItemChangedListener)
+        fragment.show(activity.supportFragmentManager, EDIT_DIALOG_TAG)
```
Comment (magiepooh) [inline demo-playground/src/main/java/com/google/android/flexbox/FlexItemClickListener.kt]: scope function 😎 

```kotlin
override fun onClick(v: View) =
      FlexItemEditFragment.newInstance(v.layoutParams as FlexItem, viewIndex)
          .apply {setFlexItemChangedListener(flexItemChangedListener)}
          .show(activity.supportFragmentManager, EDIT_DIALOG_TAG)
```

File: demo-playground/src/main/java/com/google/android/flexbox/FlexItemEditFragment.kt [hunk 33]
```diff
@@ -0,0 +1,386 @@
...
+        to.isWrapBefore = from.isWrapBefore
+    }
+
+    companion object {
+
+        private const val FLEX_ITEM_KEY = "flex_item"
+
+        private const val VIEW_INDEX_KEY = "view_index"
+
+        fun newInstance(flexItem: FlexItem, viewIndex: Int): FlexItemEditFragment {
```
Comment (magiepooh) [inline demo-playground/src/main/java/com/google/android/flexbox/FlexItemEditFragment.kt]: 😎 

```
fun newInstance(flexItem: FlexItem, viewIndex: Int): FlexItemEditFragment =
      FlexItemEditFragment().apply {
        arguments = Bundle().apply {
          putParcelable(FLEX_ITEM_KEY, flexItem)
          putInt(VIEW_INDEX_KEY, viewIndex)
        }
      }
```

Comment (MGaetan89) [inline demo-playground/src/main/java/com/google/android/flexbox/FlexItemEditFragment.kt]: In this case, we might as well remove the return type altogether:
```kotlin
fun newInstance(flexItem: FlexItem, viewIndex: Int) = FlexItemEditFragment().apply {
	arguments = Bundle().apply {
		putParcelable(FLEX_ITEM_KEY, flexItem)
		putInt(VIEW_INDEX_KEY, viewIndex)
	}
}
```

**Comment by thagikura** [comment]
Thanks for your comment, since this PR is already merged, let me address those in a follow up PR.

**Pull request merged**